File: codegen/results/v002/qwen3.5_4b_nothink/20260412_070316/go/002_word_frequency/solution.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"regexp"
	"sort"
	"strings"
)

func main() {
	file, err := os.Open("input/text.txt")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error opening file: %v\n", err)
		os.Exit(1)
	}
	defer file.Close()

	wordCount := make(map[string]int)

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()
		// Convert to lowercase
		line = strings.ToLower(line)

		// Remove punctuation: keep only letters, digits, and spaces
		// Note: The rule says "keep only letters", but typically words are separated by spaces.
		// We'll split by space, then remove non-letter characters from each word.
		pattern := regexp.MustCompile(`[^a-z]`)
		words := pattern.Split(line, -1)

		for _, word := range words {
			if word != "" {
				wordCount[word]++
			}
		}
	}

	if err := scanner.Err(); err != nil {
		fmt.Fprintf(os.Stderr, "Error reading file: %v\n", err)
		os.Exit(1)
	}

	// Create a slice of words for sorting
	type wordCount struct {
		Word   string
		Count  int
	}

	var wordList []wordCount
	for word, count := range wordCount {
		wordList = append(wordList, wordCount{Word: word, Count: count})
	}

	// Sort by count descending, then by word alphabetically ascending
	sort.Slice(wordList, func(i, j int) bool {
		if wordList[i].Count != wordList[j].Count {
			return wordList[i].Count > wordList[j].Count
		}
		return wordList[i].Word < wordList[j].Word
	})

	// Output
	for _, item := range wordList {
		fmt.Printf("%s: %d\n", item.Word, item.Count)
	}
}

File: codegen/results/v002/qwen3.5_4b_nothink/20260412_070316/typescript/002_word_frequency/solution.ts
import * as fs from 'fs';
import * as path from 'path';

const inputFile = path.resolve(process.argv[2] || 'input/text.txt');

// Read file content
const content = fs.readFileSync(inputFile, 'utf-8');

// Process the content:
// 1. Convert to lowercase
// 2. Remove all non-letter characters (keep only a-z)
// 3. Split into words
// 4. Filter out empty strings

let processedText = content.toLowerCase();

// Replace everything that is NOT a letter with nothing
processedText = processedText.replace(/[^a-z]/g, '');

// Split into words
const words = processedText.split(' ');

// Count frequencies
const wordCounts: Map<string, number> = new Map();

for (const word of words) {
  if (word.length > 0) {
    const count = wordCounts.get(word) || 0;
    wordCounts.set(word, count + 1);
  }
}

// Convert to array for sorting
const sortedWords = Array.from(wordCounts.entries())
  .sort((a, b) => {
    // Sort by count descending
    if (b[1] !== a[1]) {
      return b[1] - a[1];
    }
    // Sort alphabetically ascending for ties
    return a[0].localeCompare(b[0]);
  })
  .map(([word, count]) => `${word}: ${count}`);

// Output result
console.log(sortedWords.join('\n'));